### DISCUSSION TITLE
Built-in search in mandarin traditional chinese won't work

### DISCUSSION THREAD

**Discussion by rjjan** [fuma-nama/fumadocs#1187] [Q&A] [answered]
To Reproduce
1. use latest auto start template to create a docs project.
2. config mandarin search by this [steps] (https://fumadocs.vercel.app/docs/headless/search/orama#special-languages) in docs.
3. start dev server
4. enter some words input search input.

My api/search/route.ts
export const { GET, search } = createFromSource(source, undefined, {
  localeMap: {
    // the prop name should be its locale code in your i18n config, (e.g. `cn`)
    cn: {
      // options for the language
      tokenizer: await createTokenizer({
        stopWords: mandarinStopwords,
      }),
      search: {
        threshold: 0.1,
        tolerance: 1,
      },
    },
  },
});

question 1: using simplified chinese, won't show result, no error in the yarn dev console.  Do i need to configure something else as mentioned in the sample code?
// the prop name should be its locale code in your i18n config, (e.g. `cn`)

question 2: i need to use traditional chinese, modifying this won't give me search result either.

export const { GET, search } = createFromSource(source, undefined, {
  localeMap: {
    // the prop name should be its locale code in your i18n config, (e.g. `cn`)
    "zh": {
      // options for the language
      tokenizer: await createTokenizer({
        stopWords: mandarinStopwords,
      }),
      search: {
        threshold: 0.1,
        tolerance: 1,
      },
    },
  },
});

THANKS.

**Comment by fuma-nama** [comment]
Hmm mind to provide a reproduction (e.g. GitHub repo)?

**Reply by rjjan** [reply]
Hi Fuma,

https://github.com/rjjan/test-doc

Thanks!

**Reply by fuma-nama** [reply]
You didn't create and pass the `i18n` config to `loader`, and you can only add localized pages with locale code in their file name like `page.zh-TW.mdx`, otherwise it is tokenized without Chinese tokenizer

**Reply by rjjan** [reply]
Hi fuma-nama, 

Thanks.(I am new to Fumadocs.)
I still can't make it work
   
1. I add i18n to loader as below
2. my content/docs contains
    page.mdx
    page.zh-TW.mdx

// lib/source.ts
export const source = loader({
  baseUrl: "/docs",
  source: createMDXSource(docs, meta),
  i18n: {
    languages: ["en", "zh-TW"],
    defaultLanguage: "en",
  },
});

I Got the following errors in yarn dev console.

Error: node.children is not iterable
    at resolveErrorDev (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:1792:63)
    at processFullStringRow (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:2071:17)
    at processFullBinaryRow (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:2059:7)
    at progress (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.browser.development.js:2262:17)

**Reply by fuma-nama** [reply, marked as answer]
not sure what caused the problem, I made a working example:
https://github.com/fuma-nama/fumadocs-chinese-example
![image](https://github.com/user-attachments/assets/785ec158-96df-47d7-8f5e-d92893c9f868)

**Comment by rjjan** [comment]
Hi fuma-nama

I figure out I didn't successfully enabled i18n for the project. (I just followed the guide to "enable" the "search")
That could be the root cause.

Thank you for your example!

To share with whoever new to fumadocs like me.
1. The tutorial https://fumadocs.vercel.app/docs/ui is very good
2. for i18n enablement I suggest to follow Auto Setup guide  (https://fumadocs.vercel.app/docs/ui/internationalization#auto-setup).
3. In next 15 the params seems to be chaged to <Promise ...>
4. app/layout.config.tsx  also changed, adding  i18n: true to baseOptions

**Comment by rjjan** [comment]
Hi fuma

When yarn build,  api/search/route.ts will cause 
Type error: Route "app/api/search/route.ts" does not match the required types of a Next.js Route.
  "search" is not a valid Route export field.

By removing search from 
export const { GET, search } = createFromSource(source, undefined, {
to
export const { GET } = createFromSource(source, undefined, {

yarn build will succeed and it seems running fine.

Does the removal have any side effect ?

Thanks.

**Reply by fuma-nama** [reply]
You shouldn't export search, probably a mistake

**Comment by leovan** [comment]
Hi, @fuma-nama is there a way to tokenize a markdown file `page.md` (no `.zh-CN` suffix) with Chinese tokenizer without passing `i18n` to `loader`. Which means I want to make Chinese be the default and only language.

**Reply by fuma-nama** [reply]
Pass the tokenizer options etc to root config directly than under localMap in your `createFromSource`.

**Reply by leovan** [reply]
It works, thx.
